### DISCUSSION TITLE
SQL database with windows authentication

### DISCUSSION THREAD

**Discussion by Farman09** [nextgenhealthcare/connect#5198] [General]
Hello, 
I'm creating a Database reader channel to create a new HL7 message using Mirth Connect 4.1. What is the URL string for using windows authentication while logging into the SQL server? 
![image](https://user-images.githubusercontent.com/54945320/168946264-0e721caa-d72b-4bc7-8126-385903e672c3.png)

**Comment by Farman09** [comment]
getting the following error
![image](https://user-images.githubusercontent.com/54945320/168955260-01f7f0eb-a22a-4ff5-bebd-c62aa600e300.png)

**Reply by pacmano1** [reply]
I believe the domain should be specified in the jdbc string, not as part of the user name.

Read the FAQ/docs  at http://jtds.sourceforge.net/faq.htm

**Comment by Farman09** [comment]
Still same error even mentioning the domain

![image](https://user-images.githubusercontent.com/54945320/168958829-86d2c4f1-d27e-405f-98cb-27589171c112.png)

**Reply by pacmano1** [reply]
Your screenshot still has the domain in the username.    remove it.

**Comment by Farman09** [comment]
![image](https://user-images.githubusercontent.com/54945320/169012146-006dab9f-5d3f-4b7e-b98c-9ee1b8670105.png)
  even tried connecting using SQL authentication mode
![image](https://user-images.githubusercontent.com/54945320/169012320-98314913-be6f-4ac5-977e-def2310022fa.png)

**Reply by pacmano1** [reply]
Put an asterisk in the filter  by then get tables.

**Reply by Farman09** [reply]
Getting same error :-(

**Reply by pacmano1** [reply]
Try an actual query.    Sometimes that get tables thing doesn't work.   If the query fails post the full error message - which is normally some other connectivity thing like TLS settings.

**Reply by Farman09** [reply]
I had the following query and upon deploying the channel (after saving), 
![image](https://user-images.githubusercontent.com/54945320/169067328-4f44f2ca-154a-44c3-8760-dad8b5e4e298.png)

the channel tries to start but fails with the following error message and stops
![image](https://user-images.githubusercontent.com/54945320/169066995-a8493853-480d-453a-8905-7b043f3e787b.png)

**Comment by pacmano1** [comment]
So, google the error.   "the login is from an untrusted domain".  

Leave mirth out of the google. - it's a jdbc / jtds configuration problem.   In general when I run into problems like that I use some other jdbc compliant SQL tool to test the creds rather than futz with mirth.

**Reply by Farman09** [reply]
Still no success :-(

**Reply by pacmano1** [reply]
```
jdbc:jtds:sqlserver://<ip>:1433/dbname;domain=DOMAIN;useNTLMv2=true;ssl=request
```

**Reply by Farman09** [reply]
jdbc:jtds:sqlserver://<ip>:1433/DemoEHR;domain=FARMAN;useNTLMv2=true;ssl=request

No luck ( I have my IP address in the specified place, but not showing in the string)

**Reply by pacmano1** [reply]
What is the error?   What does the SQL server log file show?

Stop posting follow ups with "it doesn't work" without debug data and/or screenshots.

**Reply by Farman09** [reply]
My apologies. here is what the error is shown in the SQL log files
![image](https://user-images.githubusercontent.com/54945320/169142314-0b28bede-a41a-41dd-835f-c8330378473f.png)

**Reply by pacmano1** [reply]
Maybe ask a friend to help you.    

- You need collate the call that fails with the log file, not just paste a picture of logs.
- You still did not paste the most recent error when you tried to connect.
- You need to confirm that your mirth server can even open port 1433 to the SQL server.
- You need to try some product like DBSchema (which will let you test and build a jdbc connection string), you did not indicated you even tried that.

Perhaps someone else here will help,  I'm done.

**Reply by Farman09** [reply]
Thank you for your time and patience. 
**Just to UPDATE:**
The issue has been resolved and I got login success with SQL Authentication only. Still trying to figure a way to get a login with Windows Authentication, but no luck (I know you don't like when I say it). 
I have used the following string in the URL which got me success;

jdbc:jtds:sqlserver://FARMAN:1433/DemoEHR;instance=FARMAN;

**Reply by siddharthsharma9258** [reply]
I think the configuration for windows authentication need to be done on the SQL Server first and then pass connection information to Mirth.
If that config is right, Mirth won't have problem connecting to it.
